app/services/Worker.php

```diff
@@ -3,70 +3,83 @@
 namespace GanttDashboard\App\Services;
 
 use GanttDashboard\App\Models\Workers;
-use Phalcon\Session\Adapter\Files as SessionAdapter;
-use Phalcon\Security;
+use GanttDashboard\App\Validators\Worker as WorkerValidator;
+use GanttDashboard\App\Services\Authentication as AuthenticationService;
+use Phalcon\Flash\Session as FlashSession;
+use Phalcon\Mvc\View;
 
-class Worker
+class Worker extends Base
 {
     /**
-     * @var SessionAdapter
+     * @var AuthenticationService
      */
-    private $sessionService;
+    private $authenticationService;
 
     /**
-     * @var Security
+     * @var WorkerValidator
      */
-    private $securityService;
+    private $workerValidator;
 
     /**
      * @var Workers
      */
     private $workerModel;
 
     /**
-     * Constructs the needed services, set in DI, for session, security and model
-     * @param SessionAdapter $sessionService
-     * @param Security $securityService
+     * Constructs the needed services, set in DI, for session, security, model,
+     * flashSession and view
+     * @param Authentication $authenticationService
+     * @param WorkerValidator $workerValidator
      * @param Workers $workerModel
+     * @param FlashSession $flashSession
+     * @param View $view
      */
     public function __construct(
-        SessionAdapter $sessionService,
-        Security $securityService,
-        Workers $workerModel
+        AuthenticationService $authenticationService,
+        WorkerValidator $workerValidator,
+        Workers $workerModel,
+        FlashSession $flashSession,
```

Comment:
flash should also be part of the controller
if service action == success
then put something in flash

the service is just a bussiness logic that does validation and communicates with model. it is not related with views